vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/bitflyer/post_oauth/post_oauth_bitflyer.h

```diff
@@ -0,0 +1,87 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_LEDGER_SRC_BAT_LEDGER_INTERNAL_ENDPOINT_BITFLYER_POST_OAUTH_POST_OAUTH_BITFLYER_H_
+#define BRAVE_VENDOR_BAT_NATIVE_LEDGER_SRC_BAT_LEDGER_INTERNAL_ENDPOINT_BITFLYER_POST_OAUTH_POST_OAUTH_BITFLYER_H_
+
+#include <string>
+
+#include "bat/ledger/ledger.h"
+
+// POST https://bitflyer.jp/api/link/v1/token
+//
+// Request body:
+// {
+//   "client_id": "abcdedg",
+//   "client_secret": "xxxxxxxxxxxxxxxxxx",
+//   "code": "xxxxxxxxxxxxxxxxxxxxxxxxxx",
+//   "grant_type": "code",
+//   "code_verifier": "xxxxxxx",
+//   "expires_in": 3600,
```

Comment:
This doesn't appear to match the value `259002` set in `PostOauth::GeneratePayload`. Im probably just missing something here.. How are we deciding the expiration time on these requests?

Reply:
`259002` was used in bitFlyer's docs and is about 3 days. @Miyayes do we want to use a different value here?

The `3600` in the comment header was just a dummy value to show that it's a number, but I can make it match the code if it helps the clarity.

Reply:
Ive seen the 259002 in https://github.com/brave/security/issues/390 as well. It appears this can be a global constant, but are we ever going to be in a situation where it will need to be updated?

Reply:
@orspetol With the current approach, if we need to change this value in the future we'll obviously have to create a code change. In order to change it dynamically, I think we'd probably need some infrastructure support that doesn't exist quite yet (perhaps by returning a duration value with our `v1/parameters` endpoint). @Miyayes @emerick your thoughts?

Reply:
I'll ask bitFlyer if they have a preferred duration for this setting. I think if we match that, there probably won't be much need to change this in the future.

Reply:
I spoke with bitFlyer and `expires_in` is just for use on their staging environment, where it allows us to set a shorter duration to test an expired access token. In production, this value is ignored and they use a default expiration of 7 days.

cc: @Miyayes